Tests/BSONTests/BSONPointerUtilsTests.swift

```diff
@@ -0,0 +1,40 @@
+import CLibMongoC
+import Foundation
+@testable import MongoSwift
+import Nimble
+@testable import SwiftBSON
+import TestsCommon
+import XCTest
+
+final class BSONPointerUtilsTests: MongoSwiftTestCase {
```

Comment:
these tests won't really be necessary once we merge the library in and start using it, but they were helpful as sanity checks to make sure the functions I implemented actually worked.